### PR TITLE
format some markdown lists in the design doc

### PR DESCRIPTION
**Pull request by gab-umich**
Effect: it will display correctly on GitHub viewer

**Issue #, if available:**

**Description of changes:**
Removed the spaces that are blocking rendering, and reformatted some more. 

I confirm that this contribution is made under the terms of your choice and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/rfc/luthier-design.md [hunk 1]
```diff
@@ -186,38 +186,41 @@ value in Druid, but still perform human friendly searches in Fili (i.e.
...
-        Additional keys:
-            * approximateCardinality - An integer. The rough size of the dimension.
-                This is used when Fili is deciding whether to send a
-                weight-check query before sending an actual query. If not
-                present, Fili will use the query_weight_limit configuration
-                parameter.
+* fili-lucene - Uses
+    [Apache Lucene](http://lucene.apache.org/) to handle field resolution.
+
+    Additional keys:
```
Comment (archolewa) [inline docs/rfc/luthier-design.md]: Should this list be unindented as well?

File: docs/rfc/luthier-design.md [hunk 2]
```diff
@@ -186,38 +186,41 @@ value in Druid, but still perform human friendly searches in Fili (i.e.
...
+* fili-scan - Uses an in-memory data structure to handle field resolutions.
+    Not recommended for large dimensions.
+
+    Additional keys:
+        None
+
+* fili-noop - Does not do any field resolution at all. Useful for dimensions
+    that don't have any fields.
+
+    Additional keys:
```
Comment (archolewa) [inline docs/rfc/luthier-design.md]: Should this list be unindented as well?

File: docs/rfc/luthier-design.md [hunk 3]
```diff
@@ -240,18 +243,20 @@ define and use a custom key value store.
...
+* fili-reddis - Uses a Redis cluster to perform point lookups.
 
-    * fili-reddis - Uses a Redis cluster to perform point lookups.
-        Additional keys:
-            redisNamespace - The reddis namespace to use
-            storeName - The reddis store to use
-            reddisConfiguration - A table with the following keys:
-                * host - The host running the Reddis service
-                * port - The port to contact the host through. Optional.
+    Additional keys:
```
Comment (archolewa) [inline docs/rfc/luthier-design.md]: Should these be unindented?

File: docs/rfc/luthier-design.md [hunk 4]
```diff
@@ -240,18 +243,20 @@ define and use a custom key value store.
...
+    * redisNamespace - The reddis namespace to use
+    * storeName - The reddis store to use
+    * reddisConfiguration - A table with the following keys:
+    * host - The host running the Reddis service
```
Comment (archolewa) [inline docs/rfc/luthier-design.md]: `host` and `port` should use `**`. It's a sublist underneath `reddisConfiguration`

File: docs/rfc/luthier-design.md (hunk 1 above)
Comment (gab-umich) [reply docs/rfc/luthier-design.md]: A tab-indented single star list will be automatically rendered as a sublist (equiv to **) 
I think unifying style in all .md in our repo might be potentially interesting thing to do, but for this reformatting, I will make all sublist using tab + *

This is a render of what it looks like, which I think is reasonable: 
Screen Shot 2019-05-30 at 6 38 42 PM

File: docs/rfc/luthier-design.md (hunk 2 above)
Comment (gab-umich) [reply docs/rfc/luthier-design.md]: Same as above.

File: docs/rfc/luthier-design.md (hunk 3 above)
Comment (gab-umich) [reply docs/rfc/luthier-design.md]: Same as above.

File: docs/rfc/luthier-design.md (hunk 4 above)
Comment (gab-umich) [reply docs/rfc/luthier-design.md]: I have overlooked this, thank you for pointing it out.

File: docs/rfc/luthier-design.md (hunk 1 above)
Comment (archolewa) [reply docs/rfc/luthier-design.md]: 👍

**Pull request merged**